version: '3.9'

services:

    db:
        image: kartoza/postgis:14-3.2
        environment:
            - POSTGRES_DB=bndes
            - POSTGRES_USER=admin
            - POSTGRES_PASS=admin
        restart: on-failure
        healthcheck:
            test: "pg_isready -U admin -d postgres -h localhost"

    dev:
        build:
            context: .
            dockerfile: ./Dockerfile-dev
        command: >
            bash -c "export
            && python manage.py makemigrations --no-input
            && python manage.py migrate --no-input
            && python manage.py runserver 0.0.0.0:8080"
        ports:
            - "8080:8080"
        volumes:
            - .:/app/
        env_file:
            - ./.env
        environment:
            DB_USER: admin
            DB_NAME: bndes
            DB_PASS: admin
            DB_HOST: db
            PORT: 8001
            DEBUG_TKT: True
        depends_on:
            - db
        restart: on-failure

    web:
        build:
            context: .
            dockerfile: ./Dockerfile
        command: >
            bash -c "export
            && exec gunicorn --config gunicorn.conf.py connect_api.wsgi"
        ports:
            - "8000:8000"
        volumes:
            - .:/app/
        env_file:
            - ./.env
        environment:
            DB_USER: admin
            DB_NAME: bndes
            DB_PASS: admin
            DB_HOST: db
            PORT: 8000
            DEBUG_TKT: False
        restart: on-failure


